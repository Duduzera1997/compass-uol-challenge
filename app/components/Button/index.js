import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

function Button({ onPress }) {
  return (
    <SC.Container>
      <SC.TouchableButton onPress={onPress} testID="component-button-press">
        <SC.Label>Visualizar Not√≠cia Completa</SC.Label>
      </SC.TouchableButton>
    </SC.Container>
  );
}

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
};

export default Button;
