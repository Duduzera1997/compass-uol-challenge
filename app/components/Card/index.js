import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';
import ptBR from 'date-fns/locale/pt-BR';
import * as SC from './styles';

function Card({ imageUrl, title, abstract, noticeDate }) {
  return (
    <SC.ContainerCard>
      <SC.CardTouchable>
        <SC.CardContent flex={1}>
          <SC.PreviewImage source={{ uri: imageUrl }} />
        </SC.CardContent>
        <SC.CardContent flex={2.5}>
          <SC.Text size={15} fontBold>
            {title}
          </SC.Text>
          <SC.TextDate size={11.5} fontBold>
            {format(new Date(noticeDate), "dd 'de' MMMM 'de' yyyy", {
              locale: ptBR,
            })}
          </SC.TextDate>
          <SC.Text numberOfLines={2} size={13}>
            {abstract}
          </SC.Text>
        </SC.CardContent>
      </SC.CardTouchable>
    </SC.ContainerCard>
  );
}

Card.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  abstract: PropTypes.string.isRequired,
  noticeDate: PropTypes.string.isRequired,
};

export default Card;
