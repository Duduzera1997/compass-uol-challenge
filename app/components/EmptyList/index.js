import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

function EmptyList({ isLoading }) {
  return (
    <SC.ContainerEmptyList>
      {isLoading && <SC.ActivityIndicator />}
      <SC.Label>
        {isLoading
          ? 'Buscando...'
          : 'Ooops! ðŸ˜ž\n Nada foi encontrado, tente novamente...'}
      </SC.Label>
    </SC.ContainerEmptyList>
  );
}

EmptyList.defaultProps = {
  isLoading: false,
};

EmptyList.propTypes = {
  isLoading: PropTypes.bool,
};

export default EmptyList;
