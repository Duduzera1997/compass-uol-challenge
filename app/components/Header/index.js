import PropTypes from 'prop-types';
import React from 'react';

import * as SC from './styles';
import { formatDateLiteral } from '~/utils';

function Header({ headerText, dateText }) {
  return (
    <SC.ContainerHeader>
      <SC.HeaderTitle>{headerText} News</SC.HeaderTitle>
      <SC.DateLabel>Hoje Ã©, {dateText}.</SC.DateLabel>
    </SC.ContainerHeader>
  );
}

Header.defaultProps = {
  dateText: formatDateLiteral(new Date()),
};

Header.propTypes = {
  headerText: PropTypes.string.isRequired,
  dateText: PropTypes.string,
};

export default Header;
