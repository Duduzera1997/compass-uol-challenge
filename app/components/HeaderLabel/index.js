import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

function HeaderLabel({ label }) {
  return (
    <View>
      <Text>{label} News</Text>
    </View>
  );
}

HeaderLabel.defaultProps = {
  label: 'Best',
};

HeaderLabel.propTypes = {
  label: PropTypes.string,
};

export default HeaderLabel;
