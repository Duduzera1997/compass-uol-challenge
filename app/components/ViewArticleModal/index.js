import React from 'react';
import PropTypes from 'prop-types';
import { format, formatDistanceToNow } from 'date-fns';
import { Linking } from 'react-native';
import ptBR from 'date-fns/locale/pt-BR';

import * as SC from './styles';
import LinearBackground from '../LinearBackground';
import Button from '../Button';

function ViewArticleModal({ item, visible, onRequestClose }) {
  const publishedDate = new Date(item.published_date);
  return (
    <SC.ModalContainer visible={visible} onRequestClose={onRequestClose}>
      <LinearBackground>
        <SC.Container>
          <SC.ImagePreview source={{ uri: item.multimedia[0].url }} />
          <SC.ContainerIcon
            onPress={onRequestClose}
            testID="component-view-article-back-handler">
            <SC.BackIcon />
          </SC.ContainerIcon>
          <SC.Content>
            <SC.Title>{item.title}</SC.Title>
            <SC.LabelDate>
              {format(publishedDate, "dd 'de' MMMM 'de' yyyy", {
                locale: ptBR,
              })}{' '}
              | {formatDistanceToNow(publishedDate, { locale: ptBR })} atr√°s
            </SC.LabelDate>

            <SC.NoticeText>{item.abstract}</SC.NoticeText>
            <Button
              onPress={() => Linking.openURL(item.url)}
              testID="component-view-article-button-press"
            />
          </SC.Content>
        </SC.Container>
      </LinearBackground>
    </SC.ModalContainer>
  );
}

ViewArticleModal.propTypes = {
  item: PropTypes.shape({
    published_date: PropTypes.string,
    multimedia: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string,
      }),
    ),
    title: PropTypes.string,
    url: PropTypes.string,
    abstract: PropTypes.string,
  }).isRequired,
  visible: PropTypes.bool.isRequired,
  onRequestClose: PropTypes.func.isRequired,
};

export default ViewArticleModal;
