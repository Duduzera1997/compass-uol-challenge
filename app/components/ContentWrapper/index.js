import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './styles';

import Header from '../Header';
import LinearBackground from '../LinearBackground';
import SearchInput from '../SearchInput';
import ViewArticleModal from '../ViewArticleModal';

function ContentWrapper({
  title,
  children,
  showSearchInput,
  onSeachInputTextChange,
  modalProps: { visible, item, onRequestClose },
}) {
  return (
    <LinearBackground>
      <SC.Container>
        <ViewArticleModal
          visible={visible}
          item={item}
          onRequestClose={onRequestClose}
        />
        <SC.ContainerByFlex flex={1}>
          <SC.ContainerByFlex flex={1}>
            <Header headerText={title} />
          </SC.ContainerByFlex>
          {showSearchInput && (
            <SC.ContainerByFlex flex={1}>
              <SearchInput onChangeText={onSeachInputTextChange} />
            </SC.ContainerByFlex>
          )}
          <SC.ContainerByFlex flex={6}>{children}</SC.ContainerByFlex>
        </SC.ContainerByFlex>
      </SC.Container>
    </LinearBackground>
  );
}

ContentWrapper.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  showSearchInput: PropTypes.bool.isRequired,
  onSeachInputTextChange: PropTypes.func.isRequired,
  modalProps: PropTypes.shape({
    visible: PropTypes.bool.isRequired,
    item: PropTypes.shape({}),
    onRequestClose: PropTypes.func,
  }).isRequired,
};

export default ContentWrapper;
