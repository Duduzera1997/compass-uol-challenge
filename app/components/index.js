import TabIcon from './TabIcon';
import LinearBackground from './LinearBackground';
import Loading from './Loading';
import Header from './Header';
import Card from './Card';
import ListHelper from './ListHelper';
import ListArticles from './ListArticles';
import ViewArticleModal from './ViewArticleModal';
import Button from './Button';
import SearchInput from './SearchInput';
import ContentWrapper from './ContentWrapper';

export {
  TabIcon,
  LinearBackground,
  Loading,
  Header,
  Card,
  ListHelper,
  ListArticles,
  ViewArticleModal,
  Button,
  SearchInput,
  ContentWrapper,
};
