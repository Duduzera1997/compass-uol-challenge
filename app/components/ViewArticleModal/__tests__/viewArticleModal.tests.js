import React from 'react';

import { format, formatDistanceToNow } from 'date-fns';
import ptBR from 'date-fns/locale/pt-BR';
import { ViewArticleModal } from '~/components';
import { renderWithTheme, fireEvent } from '~/utils/helperTest';

const mock = {
  title: 'How to Vaccinate Homebound Seniors? Take the Shots to Them.',
  abstract:
    'Millions of older Americans, chronically ill and trapped at home, pose a big challenge to inoculation efforts. Doctors are on it.',
  url:
    'https://www.nytimes.com/2021/03/22/health/virus-seniors-vaccination.html',
  uri: 'nyt://article/e31cffee-6e8f-54eb-95dc-3ab6337fdb18',
  byline: 'By Roni Caryn Rabin and Chang W. Lee',
  item_type: 'Article',
  published_date: '2021-03-22T17:08:27-04:00',
  multimedia: [
    {
      url:
        'https://static01.nyt.com/images/2021/03/22/science/00VIRUS-HOMEBOUND-promo/00VIRUS-HOMEBOUND-promo-articleInline.jpg',
    },
  ],
};

describe('View Article Component', () => {
  it('should be contains a title send by prop', () => {
    const { getByText, toJSON } = renderWithTheme(
      <ViewArticleModal
        item={mock}
        visible={true}
        onRequestClose={jest.fn()}
      />,
    );
    expect(getByText(mock.title)).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be contains a abstract send by prop', () => {
    const onRequestClose = jest.fn();
    const { getByText, toJSON } = renderWithTheme(
      <ViewArticleModal
        item={mock}
        visible={true}
        onRequestClose={onRequestClose}
      />,
    );
    expect(getByText(mock.abstract)).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be contain a text containing the time of publication of the news up to the current time', () => {
    const { getByText, toJSON } = renderWithTheme(
      <ViewArticleModal
        item={mock}
        visible={true}
        onRequestClose={jest.fn()}
      />,
    );
    const publishedDate = new Date(mock.published_date);
    const date = `${format(publishedDate, "dd 'de' MMMM 'de' yyyy", {
      locale: ptBR,
    })} | ${formatDistanceToNow(publishedDate, { locale: ptBR })} atrÃ¡s`;
    expect(getByText(date)).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be prop onRequestClose send has been called', () => {
    const onRequestClose = jest.fn();
    const { getByTestId, toJSON } = renderWithTheme(
      <ViewArticleModal
        item={mock}
        visible={true}
        onRequestClose={onRequestClose}
      />,
    );
    const touchable = getByTestId('component-view-article-back-handler');
    fireEvent.press(touchable);
    expect(onRequestClose).toBeCalledTimes(1);
    expect(toJSON()).toMatchSnapshot();
  });

  // it('should be Button has been called', () => {
  //   const onRequestClose = jest.fn();
  //   const { getByTestId, toJSON } = renderWithTheme(
  //     <ViewArticleModal item={mock} visible={true} onRequestClose={jest.fn()} />,
  //   );
  //   const button = getByTestId('component-view-article-button-press');
  //   fireEvent.press(button);
  //   expect(button.).toBeCalledTimes(1);
  //   expect(toJSON()).toMatchSnapshot();
  // });
});
