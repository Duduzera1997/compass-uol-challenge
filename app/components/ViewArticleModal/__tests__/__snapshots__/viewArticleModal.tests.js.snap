// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Article Component should be contain a text containing the time of publication of the news up to the current time 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  visible={true}
>
  <BVLinearGradient
    colors={
      Array [
        4282066552,
        4294924895,
      ]
    }
    endPoint={
      Object {
        "x": 0,
        "y": 1.05,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://static01.nyt.com/images/2021/03/22/science/00VIRUS-HOMEBOUND-promo/00VIRUS-HOMEBOUND-promo-articleInline.jpg",
          }
        }
        style={
          Array [
            Object {
              "height": "30%",
              "opacity": 0.6,
              "width": "100%",
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 0,
            "paddingTop": 30,
            "position": "absolute",
          }
        }
        testID="component-view-article-back-handler"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 50,
              },
              Array [
                Object {},
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 20,
              "bottom": 50,
              "height": "100%",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FF5A5F",
                "fontFamily": "PTSansNarrowBold",
                "fontSize": 22,
              },
            ]
          }
        >
          How to Vaccinate Homebound Seniors? Take the Shots to Them.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#767676",
                "fontFamily": "PTSansNarrowBold",
                "fontSize": 13,
                "marginTop": 5,
              },
            ]
          }
        >
          22 de março de 2021
           
          | 
          1 dia
           atrás
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#3b2678",
                "fontFamily": "PTSansNarrowRegular",
                "fontSize": 17,
                "marginTop": "3%",
                "textAlign": "justify",
              },
            ]
          }
        >
          Millions of older Americans, chronically ill and trapped at home, pose a big challenge to inoculation efforts. Doctors are on it.
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": "10%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF5A5F",
                "borderRadius": 8,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
                "width": "70%",
              }
            }
            testID="component-button-press"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "PTSansNarrowBold",
                    "fontSize": 18,
                  },
                ]
              }
            >
              Visualizar Notícia Completa
            </Text>
          </View>
        </View>
      </View>
    </View>
  </BVLinearGradient>
</Modal>
`;

exports[`View Article Component should be contains a abstract send by prop 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  visible={true}
>
  <BVLinearGradient
    colors={
      Array [
        4282066552,
        4294924895,
      ]
    }
    endPoint={
      Object {
        "x": 0,
        "y": 1.05,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://static01.nyt.com/images/2021/03/22/science/00VIRUS-HOMEBOUND-promo/00VIRUS-HOMEBOUND-promo-articleInline.jpg",
          }
        }
        style={
          Array [
            Object {
              "height": "30%",
              "opacity": 0.6,
              "width": "100%",
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 0,
            "paddingTop": 30,
            "position": "absolute",
          }
        }
        testID="component-view-article-back-handler"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 50,
              },
              Array [
                Object {},
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 20,
              "bottom": 50,
              "height": "100%",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FF5A5F",
                "fontFamily": "PTSansNarrowBold",
                "fontSize": 22,
              },
            ]
          }
        >
          How to Vaccinate Homebound Seniors? Take the Shots to Them.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#767676",
                "fontFamily": "PTSansNarrowBold",
                "fontSize": 13,
                "marginTop": 5,
              },
            ]
          }
        >
          22 de março de 2021
           
          | 
          1 dia
           atrás
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#3b2678",
                "fontFamily": "PTSansNarrowRegular",
                "fontSize": 17,
                "marginTop": "3%",
                "textAlign": "justify",
              },
            ]
          }
        >
          Millions of older Americans, chronically ill and trapped at home, pose a big challenge to inoculation efforts. Doctors are on it.
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": "10%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF5A5F",
                "borderRadius": 8,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
                "width": "70%",
              }
            }
            testID="component-button-press"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "PTSansNarrowBold",
                    "fontSize": 18,
                  },
                ]
              }
            >
              Visualizar Notícia Completa
            </Text>
          </View>
        </View>
      </View>
    </View>
  </BVLinearGradient>
</Modal>
`;

exports[`View Article Component should be contains a title send by prop 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  visible={true}
>
  <BVLinearGradient
    colors={
      Array [
        4282066552,
        4294924895,
      ]
    }
    endPoint={
      Object {
        "x": 0,
        "y": 1.05,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://static01.nyt.com/images/2021/03/22/science/00VIRUS-HOMEBOUND-promo/00VIRUS-HOMEBOUND-promo-articleInline.jpg",
          }
        }
        style={
          Array [
            Object {
              "height": "30%",
              "opacity": 0.6,
              "width": "100%",
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 0,
            "paddingTop": 30,
            "position": "absolute",
          }
        }
        testID="component-view-article-back-handler"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 50,
              },
              Array [
                Object {},
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 20,
              "bottom": 50,
              "height": "100%",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FF5A5F",
                "fontFamily": "PTSansNarrowBold",
                "fontSize": 22,
              },
            ]
          }
        >
          How to Vaccinate Homebound Seniors? Take the Shots to Them.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#767676",
                "fontFamily": "PTSansNarrowBold",
                "fontSize": 13,
                "marginTop": 5,
              },
            ]
          }
        >
          22 de março de 2021
           
          | 
          1 dia
           atrás
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#3b2678",
                "fontFamily": "PTSansNarrowRegular",
                "fontSize": 17,
                "marginTop": "3%",
                "textAlign": "justify",
              },
            ]
          }
        >
          Millions of older Americans, chronically ill and trapped at home, pose a big challenge to inoculation efforts. Doctors are on it.
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": "10%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF5A5F",
                "borderRadius": 8,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
                "width": "70%",
              }
            }
            testID="component-button-press"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "PTSansNarrowBold",
                    "fontSize": 18,
                  },
                ]
              }
            >
              Visualizar Notícia Completa
            </Text>
          </View>
        </View>
      </View>
    </View>
  </BVLinearGradient>
</Modal>
`;

exports[`View Article Component should be prop onRequestClose send has been called 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  visible={true}
>
  <BVLinearGradient
    colors={
      Array [
        4282066552,
        4294924895,
      ]
    }
    endPoint={
      Object {
        "x": 0,
        "y": 1.05,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://static01.nyt.com/images/2021/03/22/science/00VIRUS-HOMEBOUND-promo/00VIRUS-HOMEBOUND-promo-articleInline.jpg",
          }
        }
        style={
          Array [
            Object {
              "height": "30%",
              "opacity": 0.6,
              "width": "100%",
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 0,
            "paddingTop": 30,
            "position": "absolute",
          }
        }
        testID="component-view-article-back-handler"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 50,
              },
              Array [
                Object {},
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 20,
              "bottom": 50,
              "height": "100%",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FF5A5F",
                "fontFamily": "PTSansNarrowBold",
                "fontSize": 22,
              },
            ]
          }
        >
          How to Vaccinate Homebound Seniors? Take the Shots to Them.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#767676",
                "fontFamily": "PTSansNarrowBold",
                "fontSize": 13,
                "marginTop": 5,
              },
            ]
          }
        >
          22 de março de 2021
           
          | 
          1 dia
           atrás
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#3b2678",
                "fontFamily": "PTSansNarrowRegular",
                "fontSize": 17,
                "marginTop": "3%",
                "textAlign": "justify",
              },
            ]
          }
        >
          Millions of older Americans, chronically ill and trapped at home, pose a big challenge to inoculation efforts. Doctors are on it.
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": "10%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF5A5F",
                "borderRadius": 8,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
                "width": "70%",
              }
            }
            testID="component-button-press"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "PTSansNarrowBold",
                    "fontSize": 18,
                  },
                ]
              }
            >
              Visualizar Notícia Completa
            </Text>
          </View>
        </View>
      </View>
    </View>
  </BVLinearGradient>
</Modal>
`;
